src/Microsoft.Diagnostics.EventFlow.Inputs.ActivitySource/ActivitySourceInput.cs

```diff
@@ -0,0 +1,289 @@
+                }
+
+                if (activity.Events.Any())
+                {
+                    AddPayloadProperty(e, "Events", activity.Events);
+                }
+
+                if (activity.Links.Any())
+                {
+                    AddPayloadProperty(e, "Links", activity.Links);
+                }
+            }
+
+            return e;
+        }
+
+        private void AddPayloadProperty(EventData e, string propertyName, object propertyValue)
+        {
+            e.AddPayloadProperty(propertyName, propertyValue, healthReporter_, nameof(ActivitySourceInput));
+        }
+
+        private void FlushSamplingInfoCacheIfNeeded()
+        {
+            if (activitySampling_.Count > SamplingDecisionCacheFlushThreshold)
+            {
+                lock(activitySampling_)
+                {
+                    if (activitySampling_.Count > SamplingDecisionCacheFlushThreshold)
+                    {
+                        activitySampling_.Clear();
+                    }
+                }
+            }
+        }
+
+        private void WarnAboutIneffectiveEntries(ActivitySourceInputConfiguration configuration_, IHealthReporter healthReporter)
+        {
+            // If (SomeSource, *) configuration entry is succeeded by (SomeSource, SomeActivity) configuration entry,
+            // the latter will never be used. So issue a warning.
+
+            // The presence of an entry in this dictionary means an "capture all activities" entry for a given ActivitySource
+            // has been encountered. We want to limit the warnings to one per ActivitySource, 
+            // so we flip the value from false to true after we issue a warning.
+            var captureAllActivitiesEntries = new Dictionary<string, bool>();
+
+            foreach(var source in configuration_.Sources)
+            {
+                if (string.IsNullOrWhiteSpace(source.ActivitySourceName)) continue;
+
+                bool hasAllActivityEntry = captureAllActivitiesEntries.TryGetValue(source.ActivitySourceName, out bool warningIssued);
+                if (hasAllActivityEntry)
+                {
+                    if (!warningIssued)
+                    {
+                        captureAllActivitiesEntries[source.ActivitySourceName] = true;
+
+                        healthReporter.ReportWarning(
+                            $"{nameof(ActivitySourceInput)}: configuration for source '{source.ActivitySourceName}' has an entry that applies to all activities (activity name is blank)" +
+                            " followed by other entries. The other entries will not be used. See input documentation for more information.",
+                            EventFlowContextIdentifiers.Configuration);
+                    }
+                }
+                else if (string.IsNullOrWhiteSpace(source.ActivityName))
+                {
+                    captureAllActivitiesEntries[source.ActivitySourceName] = false;
```

Comment:
This will never be updated because of line 265?

Reply:
Um, no, line 265 skips over `(*, ActivityName)` configuration. What we are checking for here is `(SourceX, *)` preceding `(SourceX, ActivityY)` configuration. 

I guess we should also warn about `(*, ActivityName)` preceding `(SourceX, ActivityName)`. I will add that warning.